src/symmetric.rs

```diff
@@ -471,6 +643,94 @@ impl SymmetricAlgorithmKey {
     }
 }
 
+#[cfg(feature = "block-cipher-trait")]
+pub use block_cipher;
+#[cfg(feature = "block-cipher-trait")]
+mod block_cipher_trait {
+    use super::*;
+
+    use core::convert::TryFrom;
+
+    use block_cipher::generic_array::{
+        typenum::{self, Unsigned},
+        ArrayLength,
+    };
+    use block_cipher::{self, Block, BlockCipher, Key, NewBlockCipher};
+
+    impl KeyBits for typenum::U128 {
+        const VALUE: Option<usize> = Some(Self::USIZE);
+    }
+    impl KeyBits for typenum::U192 {
+        const VALUE: Option<usize> = Some(Self::USIZE);
+    }
+    impl KeyBits for typenum::U256 {
+        const VALUE: Option<usize> = Some(Self::USIZE);
+    }
+
+    #[test]
+    fn assert_aes_keysize_valid() {
+        fn _inner<T: NewBlockCipher>() {}
+        _inner::<super::Key<Aes, typenum::U128>>();
+        _inner::<super::Key<Aes, typenum::U192>>();
+        _inner::<super::Key<Aes, typenum::U256>>();
+    }
+
+    impl<B: KeyBits> NewBlockCipher for super::Key<Aes, B>
+    where
+        B: typenum::Unsigned,
+        // Fancy way of allowing only {128, 192, 256}
+        B: typenum::IsGreaterOrEqual<typenum::U128, Output = typenum::B1>,
+        B: typenum::IsLessOrEqual<typenum::U256, Output = typenum::B1>,
+        B: typenum::PartialDiv<typenum::U64>,
+        // Help the trait solver see that it's also divisible by 8
+        B: typenum::PartialDiv<typenum::U8>,
+        <B as typenum::PartialDiv<typenum::U8>>::Output: ArrayLength<u8>,
+    {
+        /// Key size in bytes with which cipher guaranteed to be initialized.
+        type KeySize = <B as typenum::PartialDiv<typenum::U8>>::Output;
+
+        /// Create new block cipher instance from key with fixed size.
+        fn new(key: &Key<Self>) -> Self {
+            let prov =
+                SymmetricAlgorithm::open(SymmetricAlgorithmId::Aes, ChainingMode::Ecb).unwrap();
```

Comment:
@emgre This is where we specify using ECB. As I understand it, `BlockCipher` are meant to be used only as a block cipher primitives, which can be later wrapped by different mode structs. Because of this, I'm specifying ECB and passing `None` IV to have a 'transparent/plain' AES encryption as `BlockCipher` implementation

Reply:
Oh ok, to `BlockCipher` or whatever other crate on top of it will do the actual chaining. I get it.